src/cogent3/maths/stats/distribution.py

```diff
@@ -37,9 +37,22 @@ def zprob(x):
     return 2 * norm.sf(abs(x))
 
 
-def tprob(x, df):
+import scipy.stats
```

Comment:
imports needed to be at the top of the file. Using the linter `ruff` on the code will fix this.